### PR TITLE
Fix unbounded retry and slowdowns

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes and instructions on how this should be tested.

### CODE REVIEW

File: probe.go [hunk 1]
```diff
@@ -744,6 +745,10 @@ func (p *Probe) getNetlinkSocket() (*NetlinkSocket, error) {
 
 // attachWithKprobeEvents attaches the kprobe using the kprobes_events ABI
 func (p *Probe) attachWithKprobeEvents() error {
+	if p.kprobeEventUnavailable {
+		return fmt.Errorf("couldn't enable kprobe: kprobe_event is not available")
```
Comment (Gui774ume) [inline probe.go:749]: Actually the error is not that `kprobe_event` is not available, it's that the hook point doesn't exist. Would you mind switching the error message and attribute name ? (+ bonus points if you put the hardcoded error in `err.go`)

File: probe.go [hunk 2]
```diff
@@ -761,7 +766,11 @@ func (p *Probe) attachWithKprobeEvents() error {
 		// fallback without KProbeMaxActive
 		kprobeID, err = registerKprobeEvent(p.GetKprobeType(), p.HookFuncName, p.UID, "", p.attachPID)
 	}
+
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			p.kprobeEventUnavailable = true
```
Comment (Gui774ume) [inline probe.go:772]: See above 😄

File: probe.go [hunk 3]
```diff
@@ -480,12 +485,8 @@ func (p *Probe) init() error {
 
 	// Default retry
 	if p.ProbeRetry == 0 {
-		if p.manager.options.DefaultProbeRetry > 0 {
-			p.ProbeRetry = p.manager.options.DefaultProbeRetry
-		}
+		p.ProbeRetry = p.manager.options.DefaultProbeRetry + 1
```
[the remark this answers is not in the document]
Comment (Gui774ume) [reply probe.go:488]: We should probably add an internal field called "attempts" that we set to `1 + p.ProbeRetry` during the probe init.

[the remark this answers is not in the document]
Comment (paulcacheux) [reply probe.go:488]: The thing is setting ProbeRetry to actually mean Attemps (which was the case before with the ++) is not nice either. Maybe we should just call `retry.Do` with `retry + 1`

**Comment by Gui774ume** [comment]
@paulcacheux don't forget to reset the new internal fields in `Probe.reset()`

**Comment by paulcacheux** [comment]
Should we reset it ? Currently this is voluntary, since the hook point availability should not change, but I can at least add a comment explaining the reasoning

File: probe.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (paulcacheux) [reply probe.go:488]: Yeah I have an idea about how to this more cleanly

[the remark this answers is not in the document]
Comment (Gui774ume) [reply probe.go:488]: > Maybe we should just call retry.Do with retry + 1
👍🏻

**Review by Gui774ume — APPROVED** [review]
lgtm 👍🏻

**Pull request merged**